File: app/src/main/java/com/aero51/moviedatabase/ui/adapter/ActorImagesAdapter.java
package com.aero51.moviedatabase.ui.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView;


import com.aero51.moviedatabase.databinding.ActorImageItemBinding;
import com.aero51.moviedatabase.repository.model.tmdb.credits.ActorImagesResponse;
import com.squareup.picasso.Picasso;

import java.util.List;

import static com.aero51.moviedatabase.utils.Constants.BASE_IMAGE_URL;
import static com.aero51.moviedatabase.utils.Constants.PROFILE_SIZE_H632;

public class ActorImagesAdapter extends RecyclerView.Adapter<ActorImagesAdapter.ViewHolder> {

    private List<ActorImagesResponse.ActorImage> imagesList;


    // data is passed into the constructor
    public ActorImagesAdapter(Context context, List<ActorImagesResponse.ActorImage> imagesList) {
        this.imagesList = imagesList;
    }

    // inflates the row layout from xml when needed
    @Override
    public ActorImagesAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ActorImagesAdapter.ViewHolder(ActorImageItemBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    // binds the data to the TextView in each row
    @Override
    public void onBindViewHolder(ActorImagesAdapter.ViewHolder holder, int position) {
        ActorImagesResponse.ActorImage image  = imagesList.get(position);
        String imageUrl = BASE_IMAGE_URL + PROFILE_SIZE_H632 + image.getFile_path();
        Picasso.get().load(imageUrl).into(holder.binding.actorImageImageView);
    }

    // total number of rows
    @Override
    public int getItemCount() {
        return imagesList.size();
    }

    // stores and recycles views as they are scrolled off screen
    public class ViewHolder extends RecyclerView.ViewHolder {
        ActorImageItemBinding binding;

        ViewHolder(ActorImageItemBinding b) {
            super(b.getRoot());
            binding=b;
        }
    }

    // convenience method for getting data at click position
    private ActorImagesResponse.ActorImage getItem(int id) {
        return imagesList.get(id);
    }


}

File: app/src/main/java/com/aero51/moviedatabase/ui/adapter/ActorImagesAdapter.kt
package com.aero51.moviedatabase.ui.adapter

import android.content.Context
import com.aero51.moviedatabase.repository.model.tmdb.credits.ActorImagesResponse.ActorImage
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.aero51.moviedatabase.databinding.ActorImageItemBinding
import com.aero51.moviedatabase.utils.Constants.BASE_IMAGE_URL
import com.aero51.moviedatabase.utils.Constants.PROFILE_SIZE_H632
import com.squareup.picasso.Picasso

class ActorImagesAdapter     // data is passed into the constructor
(context: Context?, private val imagesList: List<ActorImage>) : RecyclerView.Adapter<ActorImagesAdapter.ViewHolder>() {
    // inflates the row layout from xml when needed
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(ActorImageItemBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    // binds the data to the TextView in each row
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val image =  imagesList.get(position)
        val imageUrl: String = BASE_IMAGE_URL + PROFILE_SIZE_H632 + image.file_path
        Picasso.get().load(imageUrl).into(holder.binding.actorImageImageView)
    }

    // total number of rows
    override fun getItemCount(): Int {
        return imagesList.size
    }

    // stores and recycles views as they are scrolled off screen
    inner class ViewHolder internal constructor(var binding: ActorImageItemBinding) : RecyclerView.ViewHolder(binding.root)

    // convenience method for getting data at click position
    private fun getItem(id: Int): ActorImage {

        return   imagesList.get(id)
    }
}